### DISCUSSION TITLE
Support for Mobile (Android/iOS) Syncing

### DISCUSSION THREAD

**Discussion by Gourav1910** [mahyarmirrashed/yaos#3] [Q&A] [answered]
Does we need to install anything in android mobile or just need to install the plugin and it will handle everything else? Please clarify for the non coders.

**Comment by mahyarmirrashed** [comment, marked as answer]
My original intention with this plugin was that you would have to have the basics of Git knowledge to use it. That way, if something goes awry with the plugin or with something else, you could be able to fix it yourself. If it's of value, I can create a Wiki page detailing some basic stuff you need to know and reference it in the README.md.

To answer the question of your discussion post, it will not work on Android unless you somehow have the Git binary running on that device (I haven't tried that yet, and so it's not supported). Regarding Windows Sync, I strongly encourage you not to do anything related to Git inside a Microsoft One Drive folder, Google Drive sync folder, or Windows sync. They will mess up the version management that Git is providing you. If you want to go that route, use those alternatives and disregard using this plugin. You will not have a good time mixing synchronization services.

**Comment by lucastucious** [comment]
What about implementing https://isomorphic-git.org/ for android ?

**Reply by mahyarmirrashed** [reply]
I've thought about doing mobile support through isomorphic git and I've structured the repository to be ready for it too. I was taking a look at https://github.com/denolehov/obsidian-git and they mention some performance bottle necks using it. I've been trying to look for an alternative that could potentially be better for mobile use.

If this was to be implemented, however, I can currently think of these features that would need to be added:

- Method to clone your desired vault within the plugin's setting page;
  - Handle associated potential errors with failure to clone;
  - Find out how to get people authenticated if their repositories are private and they don't have a token to use;
- Look at integrating isomorphic-git and implementing it's interface; and
- (Optional) Create automated test suite for ensuring that features/functions work as expected.

I'll say this: if there is 10 upvotes on this discussion thread, I'll get started on integrating and adding this feature for mobile to the repo.

**Reply by freenandes** [reply]
Upvoted :)

**Reply by daaanny90** [reply]
There are 14 upvotes... 😄 😄 ~~I could take a look at it, but I have never touched an obsidian plugin before, so... I think you will be faster in the implementation :)~~ If you already prepared the codebase to implement isomorphic, I think does not makes sense that somebody else with no experience touches it. But if you need help just ping me

**Reply by mahyarmirrashed** [reply]
You're right, I will get started on this soon. Just travelling the world at the moment so finding time to do this is taking a bit longer.

**Comment by thgossler** [comment]
I think this plugin has a good chance to become the most valuable free Obsidian sync solution when it would also support mobile devices.

@mahyarmirrashed, can you please have a look into the following GitHub project which is a cross-compiled git client in Web Assembly:
https://github.com/petersalomonsen/wasm-git

Could this be used instead of isomorphic-git? Perhaps it is more reliable.

**Reply by mahyarmirrashed** [reply]
This is interesting, I will look into it.

**Reply by texastoland** [reply]
Relevant: https://github.com/Vinzent03/obsidian-git/issues/572#issuecomment-1910842424
Alternative: https://github.com/joshuakto/fit/blob/master/src/fit.ts

**Comment by fetwar** [comment]
+1 a minimalistic git sync plugin that works on mobile and desktop would be terrific. If the limitations of isomorphic git can be overcome it would be substantially better than the Obsidian Git plugin in my opinion

**Reply by hilsonp** [reply]
I've been using selfhosted-livesync plugin with a self hosted couchdb for a while and the developer is awesome, full of energy and kindness but I'm not comfortable with having my documents in a database structure that I do not understand.

I've been using remotely save and it works really well with a self hosted webdav server but I'm not comfortable syncing 3 devices without conflict management.

I would love a git back-end but did not go for a git solution so far because the "every day sync" was never a click on a button which is what a mobile use require (although, I have no problem to have to spend a bit more time/energy/concentration upon rare conflict resolution).

This was until I found the joshuakto/fit project.
That was promising but it is in the early stages and is too heavy on the github api to be able to work on a 800 documents vault on a mobile. Got banned from GitHub for a day because no throttling is implemented yet.

This being said, I would love to see mobile support implemented in your more mature solution. So one more upvote from me ;-)

And thank you very much for your work !

**Comment by jestxfot** [comment]
+1
